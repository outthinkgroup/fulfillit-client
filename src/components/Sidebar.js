import React from "react"
import styled from "styled-components"

import Logo from "../elements/Logo"
import SideBarMenu from "./SideBarMenu"

const Sidebar = ({ className, isOpen }) => {
  return (
    <aside className={className}>
      <div className="container">
        <LogoSlot>
          <Logo />
        </LogoSlot>
        <SideBarMenu />
      </div>
    </aside>
  )
}

export default styled(Sidebar)`
  ${({ theme }) => theme.below.small`
    display:none;
  `}
  z-index:100;
  .container {
    box-shadow: 0px 0px 4px hsla(212, 100%, 20%, 30%);
    background: ${({ theme }) => theme.colors.lightGrey};
    position: fixed;
    padding: 20px;
    padding-top: 0;
    display: flex;
    flex-direction: column;
    height: 100vh;
    width: 267px;
  }
`
const LogoSlot = styled.div`
  margin-bottom: 50px;
  display: flex;
  align-items: center;
  ${Logo} {
    padding: 20px 0;
  }
`
