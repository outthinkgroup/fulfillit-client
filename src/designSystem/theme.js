import { css } from "styled-components";

//media breakpoints
const size = {
  small: 400,
  medium: 800,
  large: 1100
};

const theme = {
  colors: {
    primary: "#0070f3",
    darkBlue: "hsl(212, 100%, 20%)",
    lightBlue: "hsl(212, 80%, 89%)",
    lightGrey: "hsl(212, 80%, 97%)",

    warning: {
      primary: "hsl(0, 74%, 61%)",
      light: "hsl(0, 74%, 91%)",
      dark: "hsl(0, 74%, 43%)"
    },

    success: {
      primary: "hsl(140, 60%, 45%)",
      light: "hsl(140, 60%, 85%)",
      dark: "hsl(140, 60%, 29%)"
    }
  },
  depth: {
    high:
      "0 50px 100px rgba(50, 50, 93, 0.1), 0 15px 35px rgba(50, 50, 93, 0.15), 0 5px 15px rgba(0, 0, 0, 0.1)",
    medium:
      "0 25px 50px rgba(50, 50, 93, 0.05), 0 7px 17px rgba(50, 50, 93, 0.1), 0 2.5px 7.5px rgba(0, 0, 0, 0.08)",
    low:
      "0 12px 25px rgba(50, 50, 93, 0.025), 0 3px 8px rgba(50, 50, 93, 0.07), 0 1.25px 3.25px rgba(0, 0, 0, 0.06)",
    contrastLow:
      "0 12px 25px rgba(50, 50, 93, 0.025), 0 3px 8px rgba(50, 50, 93, 0.07), 0 1.25px 3.25px hsla(212, 100%, 20%, 30%)"
  },

  above: Object.keys(size).reduce((obj, key) => {
    obj[key] = (...args) => css`
      @media (min-width: ${size[key]}px) {
        ${css(...args)}
      }
    `;
    return obj;
  }, {})
};
export default theme;
