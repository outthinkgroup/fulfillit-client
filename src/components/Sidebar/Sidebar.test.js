import React from "react"
import renderer from "react-test-renderer"
import Sidebar from "./Sidebar"
import { ThemeProvider } from "styled-components"
import theme from "../../designSystem/theme"
describe("Sidebar", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Sidebar />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
