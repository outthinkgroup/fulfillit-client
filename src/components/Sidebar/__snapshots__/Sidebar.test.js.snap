// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<aside
  className="sc-bxivhb eAAlBc"
>
  <div
    className="container"
  >
    <div
      className="sc-ifAKCX dfqbbM"
    >
      <div
        className="sc-bdVaJa cVKkKw"
      >
        <span>
          fulfillit
        </span>
        <span
          className="byline"
        >
          by Out:think
        </span>
      </div>
    </div>
    <div>
      <div
        className="sc-bwzfXH ehMfMd"
      >
        <h3>
          Campaigns
        </h3>
        <nav>
          <ul>
            <li>
              <a
                href="/dashboard"
                onClick={null}
              >
                Campaigns
              </a>
            </li>
            <li>
              <a
                href="/new-campaign"
                onClick={null}
              >
                Create New Campaign
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        className="sc-bwzfXH ehMfMd"
      >
        <h3>
          Account
        </h3>
        <nav>
          <ul>
            <li>
              <a
                href="/"
                onClick={null}
              >
                User Settings
              </a>
            </li>
            <li>
              <a
                href="/"
                onClick={null}
              >
                Billing
              </a>
            </li>
            <li>
              <a
                href="/"
                onClick={null}
              >
                Account
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <span
        className="sc-htpNat eMraqZ"
      >
        <a
          className="btn"
          href="/"
          onClick={null}
        >
          Upgrade Plan
        </a>
      </span>
    </div>
  </div>
</aside>
`;
